### PR TITLE
kubectl-ownerref: use kubectl helpers

### PR DESCRIPTION
**Pull request by justinsb**
By reusing the kubectl helpers, we are able to get (relatively) easy
yaml output etc, at the expense of a little boilerplate.

### CODE REVIEW

**Comment by atoato88** [comment]
I checked the output on k8s cluster w/ [coredns-operator](https://github.com/kubernetes-sigs/cluster-addons/tree/master/coredns) and I got the result as follows:

```
$ kubectl ownerref ownerref -A CoreDNS coredns-operator                                                                     
NAMESPACE     NAME                     AGE
kube-system   serviceaccount/coredns   7m29s 

kube-system   serviceaccount/coredns-52c752mgc9   7m29s

kube-system   serviceaccount/kube-dns   7m29s

kube-system   serviceaccount/coredns   7m29s

      serviceaccount/system:coredns   3d

      serviceaccount/system:coredns   3d
```

coredns-operator creates some kinds which are Deployment, Service, ConfigMap, ... 
But output has `serviceaccount/*` kind only.
I guess [here](https://github.com/kubernetes-sigs/cluster-addons/pull/102/files#diff-ff7c1900cd4d0b89a474421f1ef512c4f5ac7c15799786ea52e34f104fe74bf7R144) is the cause point.
The kind don't change once specific `resourcePrinter` is set (ex. serviceaccount).
But you set it as [cached var expressly](https://github.com/kubernetes-sigs/cluster-addons/pull/102/files#diff-ff7c1900cd4d0b89a474421f1ef512c4f5ac7c15799786ea52e34f104fe74bf7R139-R140), so I think some updates are needed on current PR.

**Comment by atoato88** [comment]
And current command line is like as follows:
```
$ kubectl ownerref ownerref -A CoreDNS coredns-operator    
```
Double `ownerref` are needed.
Can we modify this to single `ownerref` only?

File: tools/kubectl-ownerref/main.go [hunk 1]
```diff
@@ -5,57 +5,188 @@ import (
...
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
 )
 
 func main() {
-	var kubeconfig *string
-	if homeDir := homedir.HomeDir(); homeDir != "" {
-		kubeconfig = flag.String("kubeconfig", filepath.Join(homeDir, ".kube", "config"), "kubeconfig file")
-	} else {
-		kubeconfig = flag.String("kubeconfig", "", "kubeconfig file")
+	rootCommand := cobra.Command{}
```
Comment (atoato88) [inline tools/kubectl-ownerref/main.go:30]:
```suggestion
	rootCommand := cobra.Command{}
	rootCommand.CompletionOptions.DisableDefaultCmd = true
```
`completion` sub-command exist on current code.
Is it explicitily? If not, disable `completion` sub-command by this.

Comment (justinsb) [reply tools/kubectl-ownerref/main.go:30]: Applied - thank you!

**Comment by justinsb** [comment]
Thanks for looking @atoato88 ... the kind problem in particular is tricky, because the TablePrinter seems to hack the kind in, and we need to cache it because otherwise yaml printing doesn't separate propertly.

Going to hold while I try to figure out how to fix the problems you identified - thanks again!

/hold

**Comment by justinsb** [comment]
Thinking about the `kubectl ownerref ownerref` problem, I'm wondering if a syntax like `kubectl query owner(CoreDNS:coredns)` or something of that nature might be more flexible.  But I think we can always add that as an alternative command with the same "engine" here.

I've fixed this up now so it should address the problems you identified @atoato88  - thank you!

The printing is not what I would call perfect (yaml outputs an extra `---` at the end, for example), but I think it gets us closer!

/hold cancel

File: tools/kubectl-ownerref/main.go [hunk 2]
```diff
@@ -5,60 +5,175 @@ import (
...
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *GetByOwnerOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+
+	if len(args) == 2 {
+		o.OwnerKindArg = args[0]
+		o.OwnerName = args[1]
+	} else {
+		return fmt.Errorf("syntax: kubectl ownerref [KIND] [NAME]")
```
Comment (atoato88) [inline tools/kubectl-ownerref/main.go:121]: In command line format, I think `[ ]` means optional and `< >` is better? :thinking: 

```suggestion
		return fmt.Errorf("syntax: kubectl ownerref <KIND> <NAME>")
```

File: tools/kubectl-ownerref/main.go [hunk 3]
```diff
@@ -5,60 +5,175 @@ import (
...
+
+		IOStreams: streams,
+	}
+}
+
+func NewCmdGetByOwner(parent string, f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewGetByOwnerOptions(parent, streams)
+
+	cmd := &cobra.Command{
+		Use: "kubectl ownerref",
```
Comment (atoato88) [inline tools/kubectl-ownerref/main.go:100]: Is it better to add `<KIND> <NAME>` to `Use`?
And, add example refered from [example in README.md](https://github.com/kubernetes-sigs/cluster-addons/blob/master/tools/kubectl-ownerref/README.md#usage) 

```suggestion
		Use: "kubectl ownerref <KIND> <NAME>",
		Example: `kubectl ownerref Dashboard dashboard_sample
kubectl ownerref Dashboard dashboard_sample -o yaml`,
```

**Review by atoato88 — COMMENTED** [review]
It's LGTM almost. :smile: 
I left some comments related about usage messages.

**Comment by justinsb** [comment]
Thanks @atoato88 - great points; I incorporated your suggestions!

**Comment by atoato88** [comment]
/lgtm
Thank you to incorporate them.

**Pull request merged**